ocm_python_wrapper/cluster.py

```diff
@@ -256,33 +257,96 @@ def addon_info(self):
             self.addon_name
         ).to_dict()
 
-    def validate_addon_parameters(self, parameters):
-        _info = self.addon_info()
-        _parameters = _info.get("parameters")
-        if not _parameters and parameters:
-            raise ValueError(f"{self.addon_name} does not take any parameters")
+    def filter_required_addon_parameters(self, addon_parameters):
+        """Filters out addon parameter by specified cluster-related conditions.
 
+        Args:
+            addon_parameters (dict) : Addons parameters from api
+
+        Returns: List of dicts of parameters that filtered by matching clusters current configuration and are
+        required for installation.
+        """
+        filtered_parameters = []
         required_parameters = [
-            param["id"] for param in _parameters["items"] if param["required"] is True
+            {
+                "id": param["id"],
+                "default": param.get("default_value"),
+                "conditions": param.get("conditions"),
+            }
+            for param in addon_parameters["items"]
+            if param.get("required")
         ]
-        user_addon_parameters = [param["id"] for param in parameters]
 
+        for parameter in required_parameters:
+            for param_condition in parameter["conditions"]:
+                if param_condition["resource"] == "cluster":
+                    validation_flag = True
+                    for condition, condition_value in param_condition["data"].items():
+                        cluster_value = benedict(
+                            self.instance.to_dict(), keypath_separator="."
+                        ).get(condition)
+                        # Using logical nor to check if cluster value doesn't meet all the condition requirements.
+                        if not (
+                            (
+                                isinstance(condition_value, list)
+                                and (cluster_value in condition_value)
+                            )
+                            or cluster_value == condition_value
+                        ):
+                            validation_flag = False
+                            break
+                    if validation_flag:
+                        filtered_parameters.append(parameter)
+        return filtered_parameters
+
+    def validate_addon_parameters(self, user_parameters, use_api_defaults=True):
+        """Checks user input for addon parameters.
+
+        Args:
+            user_parameters (List) : User parameter input set with values.
+            use_api_defaults (bool) : If true, set required parameter with default value to not fail as missing.
+
+        Returns: Updated parameters (if default values updated) to provide for installation.
+
+        Raises: When a required parameter is missing.
```

Comment:
```suggestion
        Raises: 
            ValueError: When a required parameter is missing.
```
also raises if params are passed and not needed

Reply:
Done